tests/test_charm_builder.py

```diff
@@ -604,18 +604,17 @@ def test_build_dependencies_virtualenv_simple(tmp_path, assert_output):
         requirements=[reqs_file],
     )
 
-    with patch("charmcraft.charm_builder._process_run") as mock:
-        with patch("shutil.copytree") as mock_copytree:
-            builder.handle_dependencies()
+    with patch("charmcraft.charm_builder.get_pip_version") as mock_pip_version:
+        mock_pip_version.return_value = (22, 0)
+        with patch("charmcraft.charm_builder._process_run") as mock:
+            with patch("shutil.copytree") as mock_copytree:
+                builder.handle_dependencies()
 
-    pip_cmd = str(charm_builder._find_venv_bin(tmp_path / STAGING_VENV_DIRNAME, "pip3"))
+    pip_cmd = str(charm_builder._find_venv_bin(tmp_path / STAGING_VENV_DIRNAME, "pip"))
 
     assert mock.mock_calls == [
         call(["python3", "-m", "venv", str(tmp_path / STAGING_VENV_DIRNAME)]),
-        call([pip_cmd, "--version"]),
-        call(
-            [pip_cmd, "install", "--upgrade", "--no-binary", ":all:", f"--requirement={reqs_file}"]
-        ),
+        call([pip_cmd, "install", f"--requirement={reqs_file}"]),
```

Comment:
So we dropped no-binary?

Reply:
The new version specifies the full `--no-binary` list by extracting it from all the requirements. Since the requirements file is empty there aren't any packages to enforce source install for, so no `--no-binary` expression gets added.